expr/shaper.go

```diff
@@ -87,6 +87,10 @@ func createOp(in, out *zng.TypeRecord) (op, error) {
 
 func (s op) castPrimitive(in zcode.Bytes, b *zcode.Builder) {
 	pc := LookupPrimitiveCaster(s.castTypes.to)
+	if in == nil {
+		b.AppendNull()
+		return
+	}
 	v, err := pc(zng.Value{s.castTypes.from, in})
```

Comment:
NIt: Keep `pc`'s definition close to its use.
```suggestion
	if in == nil {
		b.AppendNull()
		return
	}
	pc := LookupPrimitiveCaster(s.castTypes.to)
	v, err := pc(zng.Value{s.castTypes.from, in})
```